kornia/augmentation/_2d/intensity/gaussian_blur.py

```diff
@@ -7,7 +7,7 @@
 from kornia.filters import gaussian_blur2d
 
 
-class RandomGaussianBlur(IntensityAugmentationBase2D):
+class GaussianBlur(IntensityAugmentationBase2D):
```

Comment:
what I meant in the previous comments is that this entire file/function might disappear and we keep only `RandomGaussianBlur` /cc @shijianjian

Reply:
I better understand what you mean, I'll wait for you and other members decision to remove one of the two. However, if I may intervene, I still think it might be interesting to keep both as they have different purposes. Maybe clearer names for their goals should be good enough to properly manage the two in the library.

Reply:
`kornia.augmentation` operators that involves randomness -- the rest in the corresponding `kornia.*` module. I'd like to keep simple with only `RandomGaussianBlur` to reduce the number of API  to also easier maintenance. Maybe we can just describe here what what you have in mind that differ the both implementation so that we can find a good single pattern.

Reply:
Ok i'll resume below the two implementations 

- Current implementation takes a tuple as sigma to specify a fix std to construct a gaussian kernel with sigma[0] for the first axis and sigma[1] for the second axis. The randomness comes with the probability of each instance to be augmented or not.
- My proposed implementation takes a tuple as sigma to specify the range of random drawn sigma std. The drawn sigma is the same for both axis. The randomness comes with the probability of each instance to be augmented or not and that each instance is associated with a different kernel built upon the drawn sigma.

To have a single implementation for both, we could add a `sigma_mode` argument for the init module that is one of these two values respectively for both implementations:
- fix
- random

Reply:
Can't we follow the same conventions to add the `separable`? https://github.com/kornia/kornia/blob/c3fe305c6003edecc48f0d3c92f31977fcad7935/kornia/filters/gaussian.py#L10-L14

Also, I think in general:
1. Please try not change class or function names, since it might break the backward compatibility. If a name changing is required, please add deprecation warnings?
2. If changes the default class behaviour, a user warning is required to notify the users.

Reply:
I don't think separable is the right name for it as it doesn't quite describe what this PR brings. Separable allows for computing separately the axes which nothing to do with random sigma. However, I will push a version of RandomGaussianBlur with random sigma that allows for specifying separable as I empirically observed a faster computation with separate kernels.

Based on your other remarks, when we will settle on how we want the classes to be merged or if we only keep one of them, I will follow your guidance :)

Reply:
I meant to have `RandomGaussianBlur` that takes `separable` as a argument. Several scenarios here:
- If `seperable=True`, it can randomly generate x sigma and y sigma, by `sigma=(a,b)`. 
- If `seperable=True`, it can randomly generate x sigma from `(a,b)` and y sigma from `(c,d)`, by `sigma=((a,b), (c, d))`. 
- If `seperable=False`, it can randomly generate sigma, by `sigma=(a,b)`. 
- If `seperable=False`, with `sigma=((a,b), (c, d))`, an error will be thrown out.

Also, if one want to fix the sigma. He may set `sigma=(a, a)` or `sigma=((a,a),(b,b))`.

Reply:
@juliendenize i like the `sigma_mode` thing -- from my perspective I care about maintenance and health of the repo and would love to avoid ending to the follwing scenerio:
```python
from K.augmentation import GaussianBlur
from K.filters import GaussianBlur
```
that's why I vote for having two clear different api:
- `K.augmentation.RandomGaussianBlur` that will handle randomness over occurrency (`p` param) and `sigma` generation
- `K.filters.GaussianBlur` that will just implement the functional of `gaussian_blur` and no random implied.

As for the `separable` args -- to give a bit of context, we included the separable kernels basically to improve performance.

Reply:
@edgarriba @shijianjian if I understood correctly we have 3 different propositions:
- Mine: one class under `K.augmentation.RandomGaussianBlur` with `sigma_mode in ['fix', 'random']`
- @shijianjian 's: one class under `K.augmentation.RandomGaussianBlur` with the sigma mode defined by both `separable` and the input type of sigma `Union[Tuple[float,float],Tuple[Tuple[float,float],Tuple[float,float]]]`
- @edgarriba 's: one class under `K.augmentation.RandomGaussianBlur` that does what I pushed in this commit and `K.filters.GaussianBlur` that does what `K.filters.GaussianBlur2d` already does.
We can settle on one and I will do the changes accordingly, do you both have the same preference over the three choices?